driver/sqliteshim/sqliteshim.go

```diff
@@ -0,0 +1,42 @@
+// Package sqliteshim is a shim package that imports an appropriate sqlite
+// driver for the build target and registers it under ShimName.
+//
+// Currently it uses packages in the following order:
+//  • modernc.org/sqlite on supported platforms,
+//  • github.com/mattn/go-sqlite3 if Cgo is enabled,
+// Otherwise registers a driver that returns an error on unsupported platforms.
+//
+package sqliteshim
+
+import (
+	"database/sql"
+	"database/sql/driver"
+)
+
+func init() {
+	sql.Register(ShimName, shimDriver)
+}
+
+const (
+	// ShimName is the name of the shim database/sql driver registration.
+	ShimName = "sqliteshim"
+
+	// DriverName is the name of the database/sql driver. Note that
+	// the value depends on the build target.
+	DriverName = driverName
+
+	// HasDriver indicates that SQLite driver implementation is available.
+	HasDriver = !needsCgo || (needsCgo && usesCgo)
```

Comment:
I wonder why this is written in a such manner? Why not just have a bool var?

Also what do you think about changing these 3 vars to be methods? It is easier to make backwards-compatible changes this way.